server/src/main/java/org/elasticsearch/http/HttpRequestLineAndHeaders.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.http;
+
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.List;
+import java.util.Map;
+
+public interface HttpRequestLineAndHeaders {
```

Comment:
New slim interface for HTTP requests which, importantly, doesn't expose access to the contents fo the request body.

Reply:
nit: can we call this  `HttpRequestLite` or maybe  or `HttpNoBodyRequest` or maybe change the existing one to `HttpRequestWithBody` and just call this `HttpRequest` ? The existing name IMO is hard to mentally parse since request line isn't something we tend to deal with much (and this class doesn't actually expose the raw line). Also, the name limits future uses of this interface to just a line and headers.  (just a nitpick, feel free to ignore)